lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -67,18 +107,54 @@ def find_failing_method_names(actual, filter_method)
           @failing_method_names = @names.__send__(filter_method) do |name|
             @actual.respond_to?(name) && matches_arity?(actual, name)
           end
+          @failing_method_names
         end
 
         def matches_arity?(actual, name)
-          return true unless @expected_arity
+          expectation = Support::MethodSignatureExpectation.new
+
+          if @expected_arity.is_a?(Range)
+            expectation.min_count = @expected_arity.min
+            expectation.max_count = @expected_arity.max
+          else
+            expectation.min_count = @expected_arity
+          end
+
+          expectation.keywords = @expected_keywords
+          expectation.expect_unlimited_arguments = @unlimited_arguments
+          expectation.expect_arbitrary_keywords  = @arbitrary_keywords
+
+          return true if expectation.empty?
 
           signature = Support::MethodSignature.new(Support.method_handle_for(actual, name))
-          Support::StrictSignatureVerifier.new(signature, Array.new(@expected_arity)).valid?
+
+          Support::StrictSignatureVerifier.new(signature, []).with_expectation(expectation).valid?
```

Comment:
At least when I wrote this initially, `Support::StrictSignatureVerifier` was being used in `RSpec::Mocks` as well, for the use case of "does this method match these specific arguments". Before this PR, we were basically faking that by creating an array of empty items to generate our expectation. The `with_expectation(expectation)` call creates an alternate method of defining the behavior the method is supposed to match, but unless that has changed `RSpec::Mocks` still uses the other approach (which makes sense, seeing as it has a list of actual arguments it wants to check).

We probably could make the second argument optional, although that might require another PR to `RSpec::Support`.